core/src/main/java/org/apache/accumulo/core/file/blockfile/impl/CachableBlockFile.java

```diff
@@ -379,18 +381,36 @@ private BlockRead getBlock(String _lookup, BlockCache cache, BlockLoader loader)
           return new CachedBlockRead(cb, cb.getBuffer());
         }
 
+        loadLock = cache.getLoadLock(_lookup);
       }
-      /**
-       * grab the currBlock at this point the block is still in the data stream
-       *
-       */
-      _currBlock = loader.get();
 
-      /**
-       * If the block is bigger than the cache just return the stream
-       */
-      return cacheBlock(_lookup, cache, _currBlock, loader.getInfo());
+      try {
+        if (loadLock != null) {
+          loadLock.lock();
+
+          CacheEntry cb = null;
+          // check cache again after getting lock
+          cb = cache.getBlockNoStats(_lookup);
```

Comment:
Combine this with the variable declaration above.

```suggestion
          CacheEntry cb = cache.getBlockNoStats(_lookup);
```